simulator/reset/reset.go

```diff
@@ -39,17 +44,33 @@ func NewResetService(
 	deleteServices map[string]DeleteService,
 	deleteServicesForNamespacedResources map[string]DeleteServicesForNamespacedResources,
 	schedService SchedulerService,
+	ctrlerManager CtrlerManager,
 ) *Service {
 	return &Service{
 		client:                               client,
 		deleteServices:                       deleteServices,
 		deleteServicesForNamespacedResources: deleteServicesForNamespacedResources,
 		schedService:                         schedService,
+		ctrlerManager:                        ctrlerManager,
 	}
 }
 
 // Reset cleans up all resources and scheduler configuration.
 func (s *Service) Reset(ctx context.Context) error {
+	if err := s.deletek8sResources(ctx); err != nil {
+		return xerrors.Errorf("delete k8s Resources: %w", err)
+	}
+
+	// If the ResetController fail, klog.Fatalf be called internally.
+	s.ctrlerManager.ResetController()
```

Comment:
The controllers would be reset in this line. (Is an additional delete operation required?)